code/controllers/subsystem/job.dm

```diff
@@ -543,6 +543,8 @@ SUBSYSTEM_DEF(job)
 		irish_override() // Assuming direct control.
 	else if(living_mob.job == "Bartender")
 		job.give_bar_choice(living_mob, M)
+	else if(living_mob.job == "AI")
```

Comment:
check that they're a donator here, their preference will stick if they stop being a donator, and in that case they'll be stuck with what they've selected